src/zkbob/ZkBobCompoundingMixin.sol

```diff
@@ -0,0 +1,240 @@
+            : 0;
+        if (balancesDiff <= params.dust) {
+            return;
+        }
+        uint256 vaultAssets = investedAssetsAmount;
+        uint256 maxInvestedAmount = params.maxInvestedAmount;
+        if (underlyingBalance > buffer) {
+            if (vaultAssets + balancesDiff > maxInvestedAmount) {
+                balancesDiff = maxInvestedAmount - vaultAssets;
+                if (balancesDiff < minRebalanceAmount) {
+                    return;
+                }
+                investedAssetsAmount = maxInvestedAmount;
+            } else {
+                investedAssetsAmount += balancesDiff;
+            }
+            yieldVault.deposit(balancesDiff, address(this));
+            emit Rebalance(address(yieldVault), 0, balancesDiff);
+        } else {
+            if (balancesDiff > vaultAssets) {
+                balancesDiff = vaultAssets;
+                if (balancesDiff < minRebalanceAmount) {
+                    return;
+                }
+                investedAssetsAmount = 0;
+            } else {
+                investedAssetsAmount -= balancesDiff;
+            }
+            yieldVault.withdraw(balancesDiff, address(this), address(this));
+            emit Rebalance(address(yieldVault), balancesDiff, 0);
+        }
+    }
+
+    /**
+     * @dev Collects accumulated fees and generated yield for the specific collateral.
+     * Callable only by the contract owner / proxy admin / yield admin.
+     * @param minClaimAmount minimum amount of token to claim.
+     * @return Claimed amount.
+     */
+    function claim(uint256 minClaimAmount) external returns (uint256) {
+        YieldParams storage params = yieldParams;
+
+        address yieldAddress = params.yield;
+
+        if (yieldAddress == address(0)) {
+            return 0;
+        }
+
+        address operator = params.yieldOperator;
+        require(
+            operator == address(0) || operator == msg.sender, "ZkBobCompoundingPool: Claim is an operator-called method"
+        );
+
+        uint256 dust = params.dust;
+        minClaimAmount = minClaimAmount > dust ? minClaimAmount : dust;
+
+        return _claim(minClaimAmount, yieldAddress, params.interestReceiver, dust);
+    }
+
+    function emergencyWithdraw() external onlyOwner {
+        YieldParams storage params = yieldParams;
+
+        address yieldAddress = params.yield;
+
+        if (yieldAddress == address(0)) {
+            return;
+        }
+
+        IERC4626 yieldVault = IERC4626(yieldAddress);
+
+        uint256 currentInvestedSharesAmount = yieldVault.balanceOf(address(this));
+        uint256 allAssets = yieldVault.convertToAssets(currentInvestedSharesAmount);
+        uint256 currentInvestedAssetsAmount = investedAssetsAmount;
+        uint256 rebalancedAssets = allAssets;
+
+        yieldVault.withdraw(allAssets, address(this), address(this));
+        if (allAssets > currentInvestedAssetsAmount) {
```

Comment:
This comparation can be inaccurate, better to compare amount after real withdrawa